csvutil/csv.go

```diff
@@ -27,6 +27,34 @@ func min(a, b int) int {
 	return b
 }
 
+func formatValue(val interface{}) (string, error) {
+	var err error
+	rv := reflect.Indirect(reflect.ValueOf(val))
+	rt := rv.Type()
+	switch rt.Kind() {
+	case reflect.Bool:
+		return strconv.FormatBool(rv.Bool()), nil
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+		return strconv.FormatInt(rv.Int(), 10), nil
+	case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+		return strconv.FormatUint(rv.Uint(), 10), nil
+	case reflect.Float32, reflect.Float64:
+		return strconv.FormatFloat(rv.Float(), 'g', -1, rt.Bits()), nil
+	case reflect.String:
+		return rv.String(), nil
+	case reflect.Slice:
+		vals := make([]string, rv.Len())
```

Comment:
perhaps use a `string.Builder` instead? (and take it as an argument to `formatValue`?)

Reply:
done!

Reply:
not passed to formatValue though, I'm not sure how that would work...